contracts/src/v0.8/automation/upkeeps/LinkAvailableBalanceMonitor.sol

```diff
@@ -35,94 +35,76 @@ interface ILinkAvailable {
-  /// @param topUpAmount the amount of LINK to top up an aggregator with at once
-  constructor(
-    address linkTokenAddress,
-    uint256 topUpAmount,
-    uint16 maxPerform,
-    uint16 maxCheck
-  ) ConfirmedOwner(msg.sender) {
+  constructor(address linkTokenAddress, uint16 maxPerform, uint16 maxCheck) ConfirmedOwner(msg.sender) {
     require(linkTokenAddress != address(0), "LinkAvailableBalanceMonitor: invalid linkTokenAddress");
-    require(topUpAmount > 0, "LinkAvailableBalanceMonitor: invalid topUpAmount");
     LINK_TOKEN = IERC20(linkTokenAddress);
-    s_topUpAmount = topUpAmount;
-    s_maxPerform = maxPerform;
-    s_maxCheck = maxCheck;
+    setMaxPerform(maxPerform);
+    setMaxCheck(maxCheck);
   }
 
   /// @notice Sets the list of subscriptions to watch and their funding parameters
   /// @param addresses the list of target addresses to watch (could be direct target or IAggregatorProxy)
-  /// @param minBalances the list of corresponding minBalance for the target address
-  function setWatchList(address[] calldata addresses, uint256[] calldata minBalances) external onlyOwner {
-    if (addresses.length != minBalances.length) {
+  /// @param minBalancesWei the list of corresponding minBalance for the target address
+  /// @param topUpAmountsWei the list of corresponding minTopUp for the target address
+  function setWatchList(
+    address[] calldata addresses,
+    uint256[] calldata minBalancesWei,
+    uint256[] calldata topUpAmountsWei
+  ) external onlyOwner {
+    if (addresses.length != minBalancesWei.length || addresses.length != topUpAmountsWei.length) {
       revert InvalidWatchList();
     }
-    // first, remove all existing addresses from list
-    for (uint256 idx = s_watchList.length(); idx > 0; idx--) {
-      (address target, ) = s_watchList.at(idx - 1);
-      require(s_watchList.remove(target), "LinkAvailableBalanceMonitor: unable to setWatchlist");
+    address[] memory oldWatchList = s_watchList;
+    for (uint256 idx = 0; idx < oldWatchList.length; idx++) {
+      address target = oldWatchList[idx];
+      delete s_targets[target];
     }
-    // then set new addresses
     for (uint256 idx = 0; idx < addresses.length; idx++) {
-      if (s_watchList.contains(addresses[idx])) {
+      if (s_targets[addresses[idx]].isActive == true) {
         revert DuplicateAddress(addresses[idx]);
       }
-      if (addresses[idx] == address(0)) {
+      if (topUpAmountsWei[idx] < 0) {
         revert InvalidWatchList();
       }
-      s_watchList.set(addresses[idx], minBalances[idx]);
-    }
-    emit WatchlistUpdated();
-  }
-
-  /// @notice Adds addresses to the watchlist without overwriting existing members
-  /// @param addresses the list of target addresses to watch (could be direct target or IAggregatorProxy)
-  /// @param minBalances the list of corresponding minBalance for the target address
-  function addToWatchList(address[] calldata addresses, uint256[] calldata minBalances) external onlyOwner {
```

Comment:
Why are we removing `addToWatchList` and `removeFromWatchList`?

Reply:
On one hand, I migrated s_watchlist from EnumerableMap to an array, since we have the need now to store at least minBalance and topUpAmount per address we can't no longer use a map.
With EnumerableMap it makes sense (and it's easy) to maintain add and remove, with an array it's much more difficult and can led us to further bugs. 
And on the other hand, those functions apparently hasn't been used at all, all the work is done by setWatchlist.